packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard.tsx

```diff
@@ -0,0 +1,230 @@
+import { NFTCardStyledAssetPlayer, TokenIcon } from '@masknet/shared'
+import { makeStyles } from '@masknet/theme'
+import { Alchemy_EVM, RSS3BaseAPI } from '@masknet/web3-providers'
+import { NetworkPluginID } from '@masknet/web3-shared-base'
+import { ChainId, resolveIPFSLinkFromURL, ZERO_ADDRESS } from '@masknet/web3-shared-evm'
+import { Box, Typography, Card } from '@mui/material'
+import differenceInCalendarDays from 'date-fns/differenceInDays'
+import differenceInCalendarHours from 'date-fns/differenceInHours'
+import { useMemo } from 'react'
+import { ReversedAddress } from './ReversedAddress'
+import { useI18N } from '../../locales'
+import { useAsyncRetry } from 'react-use'
+
+const useStyles = makeStyles()((theme) => ({
+    wrapper: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        marginBottom: 16,
+        cursor: 'pointer',
+    },
+    img: {
+        width: '64px !important',
+        height: '64px !important',
+        borderRadius: '8px',
+        objectFit: 'cover',
+    },
+    collection: {
+        borderLeft: `4px solid ${theme.palette.maskColor.line}`,
+        paddingLeft: 12,
+        marginTop: 12,
+        marginLeft: 8,
+    },
+    time: {
+        color: theme.palette.maskColor.third,
+        marginLeft: 10,
+    },
+    summary: {
+        textOverflow: 'ellipsis',
+        '-webkit-line-clamp': '1',
+        maxWidth: '400px',
+        overflow: 'hidden',
+        display: '-webkit-box',
+        '-webkit-box-orient': 'vertical',
+    },
+    defaultImage: {
+        background: theme.palette.maskColor.modalTitleBg,
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        borderRadius: 12,
+    },
+    loadingFailImage: {
+        minHeight: '0 !important',
+        maxWidth: 'none',
+        width: 64,
+        height: 64,
+    },
+}))
+
+export const ChainID = {
+    ethereum: ChainId.Mainnet,
+    polygon: ChainId.Matic,
+    bnb: ChainId.BSC,
+}
+
+export interface FeedCardProps {
+    feed: RSS3BaseAPI.Web3Feed
+    address?: string
+    onSelect: (feed: RSS3BaseAPI.Web3Feed) => void
+}
+
+export function FeedCard({ feed, address, onSelect }: FeedCardProps) {
+    const { classes } = useStyles()
+    const t = useI18N()
+
+    const { value: NFTMetadata } = useAsyncRetry(async () => {
+        if ((feed.title && feed.summary) || !feed.metadata?.collection_address) return
+
+        const res = await Alchemy_EVM.getAsset(feed.metadata?.collection_address, feed.metadata?.token_id ?? '', {
+            chainId: ChainID[feed.metadata?.network ?? 'ethereum'],
+        })
+        return res
+    }, [feed.metadata?.collection_address])
+
+    const action = useMemo(() => {
+        if (!feed) return
+        if (feed.tags?.includes('NFT')) {
+            if (feed.metadata?.from?.toLowerCase() === address) {
```

Comment:
Please always use `isSameAddress` `isZeroAddress` `isNativeTokenAddress` for tokens' address comparison.